### PR TITLE
Upgrade HealthChecks to netstandard 2.1 and UI and samples to netcoreapp 3.1

### PR DESCRIPTION
**Pull request by CarlosLanderas**
**What this PR does / why we need it**:

Target to latest netstandard and netcoreapp versions.
The new healthchecks nuget versions will be 3.1.X

**Which issue(s) this PR fixes**:

Please reference the issue this PR will close: #467

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [X] Code compiles correctly
- [X] Unit tests passing
- [X] End-to-end tests passing

### CODE REVIEW

**Comment by CarlosLanderas** [comment]
Blocked until @unaizorrilla finishes some pending work so we can freeze 3.0 healthchecks branch

File: README.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ HealthChecks packages include health checks for:
 - SignalR
 - Kubernetes
 
-> We support netcoreapp 2.2 and the new netcoreapp 3.0. Please use 2.2.X package version for .NET Core 2.2 and 3.0.X for .NET Core 3.0
+> We support netcoreapp 2.2, 3.0 and 3.1, Please use package versions 2.2.X, 3.0.X and 3.1.X to target different versions.
```
Comment (sungam3r) [inline README.md]:
```suggestion
> We support netcoreapp 2.2, 3.0 and 3.1. Please use package versions 2.2.X, 3.0.X and 3.1.X to target different versions.
```

**Comment by vip32** [comment]
why was this netstandard2.1 upgrade needed? are any new features of the new 2.1 used?
the issue is that now you enforce all packages using yours to upgrade to 2.1, which where fine on 2.0 before. also netcoreapp3.1 works fine with netstandard2.0 packages

**Comment by unaizorrilla** [comment]
Hi @vip32 
There is any issue related with this upgrade? Is this related with full fx support? Can you explain more why this is a bad idea?

**Comment by vip32** [comment]
only upgrade to 2.1 if any of the new features are needed, otherwise stick with 2.0.
even microsoft sticks to this. not sure if this is a general guideline, but it makes sense.

**Comment by unaizorrilla** [comment]
Let us discuss this!

**Comment by vip32** [comment]
yes please, and listen to Andrew a bit :)

"_By continuing to target .NET Standard 2.0, you will be able to consume it in .NET Core 3.0 applications, but you'll also continue to be able to consume it in .NET Core 2.x applications,_"

"_By targeting .NET Standard 2.0, you're allowing a large number of frameworks to consume your library. Upgrading to .NET Standard 2.1 will limit that significantly._ "

https://andrewlock.net/converting-a-netstandard-2-library-to-netcore-3/

**Comment by unaizorrilla** [comment]
> 
> 
> yes please, and listen to Andrew a bit :)
> 
> "_By continuing to target .NET Standard 2.0, you will be able to consume it in .NET Core 3.0 applications, but you'll also continue to be able to consume it in .NET Core 2.x applications,_"
> 
> "_By targeting .NET Standard 2.0, you're allowing a large number of frameworks to consume your library. Upgrading to .NET Standard 2.1 will limit that significantly._ "
> 
> https://andrewlock.net/converting-a-netstandard-2-library-to-netcore-3/

We listen Andrew always!

**Comment by vip32** [comment]
or at least multitarget 2.0 + 2.1

**Comment by CarlosLanderas** [comment]
@vip32 , no need to multitarget as we do not uses any feature from 2.1 and you are right. Let me upgrade the packages back to 2.0

**Comment by vip32** [comment]
nice! many consumers of your great packages will be happy :)

**Comment by CarlosLanderas** [comment]
Sometimes we are more on the development side than the user side. This kind of feeback is appreciated. Thanks :)

**Comment by vip32** [comment]
no probs, we see more projects having this challenge

**Comment by unaizorrilla** [comment]
Hi @vip32 

Thanks again for point us this issue!! We try to fix asap!

**Comment by CarlosLanderas** [comment]
@vip32, the change have been merged and is in the appveyor queue to be published.

image

**Pull request merged**